unit-testing/presentation.html

```diff
@@ -1,27 +1,310 @@
+        </section>
+      </section>
+      <!-- 13 slide -->
+      <section>
+        <h3 class="slide-name">Why Jest?</h3>
+        <ul>
+          <li class="fragment">Facebook developed the Jest unit testing framework.</li>
+          <li class="fragment">The default template of React Native ships with Jest testing framework.</li>
+        </ul>
+      </section>
+      <!-- 14 slide -->
+      <section>
+        <h3 class="slide-name">How to set up Jest?</h3>
+        <p class="font-size-2 align-center">
+          After command ​'react-native init' The following configuration should be automatically added to your
+          package.json file:</p>
+        <pre><code data-trim data-noescape>
+            {​​
+              "scripts": {​​
+                "test": "jest",​​
+              },​​
+              "jest": {​​
+                "preset": "react-native",​​
+              }​​
+            }
+          </code></pre>
+      </section>
+      <!-- 15 slide -->
+      <section>
+        <h3 class="slide-name">jest.config.js and jest.setup.js files​</h3>
+        <p class="font-size-2 align-center">
+          A list of paths to modules that run some code to configure or set up the testing environment. Each setupFile
+          will be run once per test file.​​​</p>
+        <pre><code data-trim data-noescape>
+            module.exports = {​​
+              roots: ['<rootDir>/src'],​​
+              collectCoverage: true,​​
+              coverageThreshold: {​​ ... },​​
+              preset: 'react-native',​​
+              setupFiles: ['<rootDir>jest.setup.js'],​​
+              setupFilesAfterEnv: ['<rootDir>/src/globalMock.js'],​​
+            };
+          </code></pre>
+      </section>
+      <!-- 16 slide -->
+      <section>
+        <h3 class="slide-name">Structuring Tests​</h3>
+        <pre><code data-trim data-noescape>
+          beforeAll(() => {​
+            // ...​​
+          });​​
+          afterAll(() => {​​
+            // ...​​
+          });​​
+          describe('Our component', () => {​​
+            beforeEach(() => {​​
+             // ...​​
+            });​​
+            it('should test this', () => {​​
+             // ...​​
+            });​​
+          });
+          </code></pre>
+      </section>
+      <!-- 17 slide -->
+      <section>
+        <h3 class="slide-name">Expect​</h3>
+        <pre><code data-trim data-noescape>
+          expect(value)​​
+          expect.anything()​​
+          expect.any(constructor)​​
+          .not​​
+          .toBe(value)​​
+          .toHaveBeenCalled()​​
+          .toBeLessThan(number | bigint)​​
+          .toBeNull()​​
+          .toContain(item)​​
+          .toEqual(value)​​
+          .toMatch(regexp | string)​​
+          .toMatchSnapshot(propertyMatchers?, hint?)​​
+          .toThrow(error?)​​
+        </code></pre>
+      </section>
+      <!-- 18 slide -->
+      <section>
+        <h3 class="slide-name">Mocking & The Jest Object​</h3>
```

Comment:
could you add few words about why we need to use mocks?
stubs VS mocks?

Reply:
done